qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/AbstractRollingTestCase.java

```diff
@@ -31,6 +34,16 @@ public static ClusterType parse(String value) {
     protected static final boolean FIRST_MIXED_ROUND = Boolean.parseBoolean(System.getProperty("tests.first_round", "false"));
     protected static final Version UPGRADE_FROM_VERSION = Version.fromString(System.getProperty("tests.upgrade_from_version"));
 
+    protected static IndexVersion getOldClusterIndexVersion() {
+        var version = UPGRADE_FROM_VERSION;
+        if (version.equals(org.elasticsearch.Version.CURRENT)) {
```

Comment:
Shouldn't this be comparing with a fixed constant instead of current?

Reply:
It's basically a shortcut to map `Version.CURRENT` to `IndexVersion.current()`, as there are some test runs where that's needed. It's the same in `ParameterizedFullClusterRestartTestCase`

Reply:
The else case here asserts that the bwc version is less than 8.11. What happens when we bump main (and therefore CURRENT) to 8.12? I think that that assertions would trip.